frontend/english-keyset.ts

```diff
@@ -0,0 +1,339 @@
+        ],
+        ["Tab", "Q", "W", "E", "R", "T", "Y", "U", "I", "O", "P", "{", "}", "|"],
+        ["Caps", "A", "S", "D", "F", "G", "H", "J", "K", "L", ":", '"', "Enter"],
+        ["Shift", "Z", "X", "C", "V", "B", "N", "M", "<", ">", "?", "Shift"],
+        ["Ctrl", "Alt", "Space", "Alt", "Ctrl"],
+    ];
+
+    constructor() {
+        super();
+        this.languageName = "English";
+
+        this.unShiftedKeys = this.unshiftedKeyStrs.map((row) => {
+            let keys: Key[] = [];
+            row.forEach((key_str) => {
+                let span = this.getKeySpan(key_str);
+                let is_special = this.isSpecialKey(key_str);
+                let onDownFunction = this.getKeyDownFunction(key_str);
+                let onUpFunction = this.getKeyUpFunction(key_str);
+                let onMouseDownFunction = this.getMouseDownFunction(key_str);
+                let onMouseUpFunction = this.getMouseUpFunction(key_str);
+
+                keys.push(
+                    new Key(
+                        "",
+                        key_str,
+                        "",
+                        1,
+                        is_special,
+                        span,
+                        onDownFunction,
+                        onUpFunction,
+                        onMouseDownFunction,
+                        onMouseUpFunction,
+                    ),
+                );
+            });
+            return keys;
+        });
+
+        this.shiftedKeys = this.shiftedKeyStrs.map((row) => {
+            let keys: Key[] = [];
+            row.forEach((key_str) => {
+                let span = this.getKeySpan(key_str);
+                let is_special = this.isSpecialKey(key_str);
+
+                let onDownFunction = this.getKeyDownFunction(key_str);
+                let onUpFunction = this.getKeyUpFunction(key_str);
+                let onMouseDownFunction = this.getMouseDownFunction(key_str);
+                let onMouseUpFunction = this.getMouseUpFunction(key_str);
+
+                keys.push(
+                    new Key(
+                        "",
+                        key_str,
+                        "",
+                        1,
+                        is_special,
+                        span,
+                        onDownFunction,
+                        onUpFunction,
+                        onMouseDownFunction,
+                        onMouseUpFunction,
+                    ),
+                );
+            });
+            return keys;
+        });
+    }
+
+    private getKeyDownFunction(key_str: string) {
+        return (e: KeyboardEvent) => { console.log(key_str + " Key Down") };
+    }
+
+    private getKeyUpFunction(key_str: string) {
+        return (e: KeyboardEvent) => { console.log(key_str + " Key Up") };
+    }
+
+    private getMouseDownFunction(key_str: string) {
+
+        let customMouseDownFn = () => {
+
+            if (key_str.toLowerCase() in ["shift", "ctrl", "alt"]) {
```

Comment:
could you capture <kbd>Caps Lock</kbd> as well?

Reply:
Caps lock is not in the scope of this PR.

I'll capture it in the next one.